pkg/fixedpoint/convert.go

```diff
@@ -227,11 +229,11 @@ func (v Value) IsZero() bool {
 }
 
 func Mul(x, y Value) Value {
-	return NewFromFloat(x.Float64() * y.Float64())
+	return mulOp(x, y)
```

Comment:
This might make the mul func slower since there's additional function call. Can you do some bench on the changes to convince people that this works better? Also it's rare that we need to multiply an integer.